src/FSharpVSPowerTools.Logic/VSLanguageService.fs

```diff
@@ -35,6 +37,18 @@ and IProjectProvider =
 
 type ShowProgress = OperationState -> unit
 
+type EntityCache() =
+    let dic = Dictionary<AssemblyPath, AssemblyContentCacheEntry>()
+    interface IAssemblyContentCache with
+        member __.TryGet assembly =
+            match dic.TryGetValue assembly with
+            | true, entry -> Some entry
+            | _ -> None
+        member __.Set assembly entry = dic.[assembly] <- entry
+
+    member __.Clear() = dic.Clear()
+    member x.Locking f = lock dic <| fun _ -> f (x :> IAssemblyContentCache)
```

Comment:
Can we move the lock to the `Set` function and avoid passing `Locking` member around?